### PR TITLE
write columnmetadata to the behind of the column chunk data, not the ColumnChunk

### LINKED ISSUE #1946: Write error ColumnChunk to the Parquet File instead of ColumnMetaData
**Describe the bug**
I'm working on the https://github.com/apache/arrow-rs/pull/1935, and go through the write path of rust-version.

I find that parquet-write write some error data which is ColumnChunk to the file, you can find the logic from https://github.com/apache/arrow-rs/blob/9f7b6004d365b0c0bac8e30170b49bdd66cc7df0/parquet/src/file/writer.rs#L441

From my knowledge about parquet format in this https://github.com/apache/parquet-format/blob/54e53e5d7794d383529dd30746378f19a12afd58/src/main/thrift/parquet.thrift#L789 and https://github.com/apache/parquet-format#file-format, the data after column data is  `ColumnMetaData`, not ColumnChunk

**To Reproduce**

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

Closes #1946

# Rationale for this change
 
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by liukun4515** [comment]
@tustvold PTAL

File: parquet/src/file/metadata.rs [hunk 1]
```diff
@@ -611,6 +611,29 @@ impl ColumnChunkMetaData {
             encrypted_column_metadata: None,
         }
     }
+
+    /// Method to convert to Thrift `ColumnMetaData`
+    pub fn to_column_metadata_thrift(&self) -> ColumnMetaData {
```
Comment (tustvold) [inline parquet/src/file/metadata.rs:616]: Could we change `to_thrift` above to use this method

File: parquet/src/file/writer.rs [hunk 2]
```diff
@@ -435,12 +435,15 @@ impl<'a, W: Write> SerializedPageWriter<'a, W> {
         Ok(self.sink.bytes_written() - start_pos)
     }
 
-    /// Serializes column chunk into Thrift.
+    /// Serializes column metadata into Thrift.
     /// Returns Ok() if there are not errors serializing and writing data into the sink.
     #[inline]
-    fn serialize_column_chunk(&mut self, chunk: parquet::ColumnChunk) -> Result<()> {
+    fn serialize_column_chunk(
```
Comment (tustvold) [inline parquet/src/file/writer.rs:441]: Could we just remove this method and move its content into write_metadata?

**Review by tustvold — COMMENTED** [review]
I think this makes sense, I left some minor nits. I think with a simple unit test to make sure the data, this can go in.

FWIW I did some digging and tbh I'm not actually sure if anything actually ever reads this data, parquet-mr doesn't even write it, arrow C++ writes it but doesn't appear to ever read it (https://github.com/apache/parquet-cpp/pull/224). Hive and Impala allegedly makes use of it, but I can't find out where/how/why...

I can't help wondering if this was an oversight in the original parquet specification, not collocating column chunk metadata in the footer, that has since been papered over

**Comment by liukun4515** [comment]
> I can't help wondering if this was an oversight in the original parquet specification, not collocating column chunk metadata in the footer, that has since been papered over. All readers I can find simply read the ColumnChunkMetadata from the footer and ignore everything else.

I have the same confuse like you about the meatdata.
I go through the parquet-mr(Java version) which did't append this metadata in end of each column, and read this metadata from the Filemetadata in the footer.

But from the definition of the format https://github.com/apache/parquet-format/blob/54e53e5d7794d383529dd30746378f19a12afd58/src/main/thrift/parquet.thrift#L790, we can know the `file_offset` is required field and the https://github.com/apache/parquet-format/blob/54e53e5d7794d383529dd30746378f19a12afd58/src/main/thrift/parquet.thrift#L796 ColumnMetaData is a optional field.

File: parquet/src/file/metadata.rs (hunk 1 above)
Comment (liukun4515) [reply parquet/src/file/metadata.rs:616]: good catch

File: parquet/src/file/writer.rs (hunk 2 above)
Comment (liukun4515) [reply parquet/src/file/writer.rs:441]: I don't have any preferences

**Comment by liukun4515** [comment]
Many system or reader just read the footer and get the metadata, I think we should just follow the parquet-format.
Maybe it's just historical issues or historical design

**Review by tustvold — APPROVED** [review]

**Pull request merged**